### ISSUE TITLE
pillar files not named init.sls

### ISSUE THREAD

**Issue by jeffdyke** [saltstack/salt#47315]
### Description of Issue/Question
Just a simple question, that i can't seem to find the answer to.  

I'm using `git-crypt` to protect files in pillars, and want to following naming convention for the glob of which should be encrypted and which should not, on the salt master the repository is unlocked, so no issue there.  I've tried using dashes like `init-locked.sls` or just `locked.sls`, but each variation returns the same response: 
` Specified SLS 'foo' in environment 'bar' is not available on the salt master`  where `foo` is a directory with only one file `locked.sls`

I've found older various issues where people have tried to put `.`'s in their filenames, which is why i used a dash or just another word, but neither are getting picked up
### Setup
Nothing special here.  I can do this another way and use the directory name as the glob, but was just curious about the issue.

### Steps to Reproduce Issue
Add a directory and file to pillar data, not called `init.sls` 
run `salt "*" saltutil.resfresh_pillar`
run `salt minion_id pillar.items | less`

You should see the error right at the top

### Versions Report
Salt Version:
           Salt: 2018.3.0

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Dec  4 2017, 14:50:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-1055-aws
         system: Linux
        version: Ubuntu 16.04 xenial

**Comment by garethgreenaway** [comment]
@jeffdyke Thanks for the report.  What does the top file for your pillar data look like?

**Comment by jeffdyke** [comment]
@garethgreenaway - looks like out posts crossed, but to answer you question, that goes along with my comment, it now looks like this

```
devbuilds:
  'environment:devbuilds':
    - match: grain
    - mariadb-users
    - mariadb-grants
    - play
    - cron-jobs
    - permissions
    - servers
    - flyway.locked
    - google
    - nginx
```
where `flyway` directory now has both an `init.sls` file, which is empty and `locked.sls` which has the requisite yaml

**Comment by garethgreenaway** [comment]
@jeffdyke That is correct.  Any directories under `/srv/pillar` require an init.sls, even an empty one , to ensure that the directory is available.
